dokka-integration-tests/gradle/src/testExampleProjects/kotlin/ExampleProjectsTest.kt

```diff
@@ -0,0 +1,344 @@
+        }
+    }
+
+    /**
+     * Identifier for each example project, to avoid string comparisons everywhere.
+     */
+    enum class ExampleProject {
+        BasicGradle,
+        CompositeBuild,
+        CustomStyling,
+        Java,
+        Javadoc,
+        KotlinAsJava,
+        KotlinMultiplatform,
+        LibraryPublishing,
+        Multimodule,
+        VersioningMultimodule,
+        ;
+
+        companion object {
+            fun of(dir: Path): ExampleProject {
+                return when (dir.name) {
+                    "basic-gradle-example" -> BasicGradle
+                    "javadoc-example" -> Javadoc
+                    "composite-build-example" -> CompositeBuild
+                    "custom-styling-example" -> CustomStyling
+                    "java-example" -> Java
+                    "kotlin-as-java-example" -> KotlinAsJava
+                    "kotlin-multiplatform-example" -> KotlinMultiplatform
+                    "library-publishing-example" -> LibraryPublishing
+                    "multimodule-example" -> Multimodule
+                    "versioning-multimodule-example" -> VersioningMultimodule
+                    else -> error("Undeclared example project: $dir")
+                }
+            }
+        }
+    }
+
+
+    @ParameterizedTest
+    @ArgumentsSource(TestCaseProvider::class)
+    fun `test HTML output`(testCase: TestCase) {
+        assumeTrue(testCase.outputsHtml)
+
+        testDokkaOutput(
+            testCase = testCase,
+            format = "html",
+        )
+    }
+
+    @ParameterizedTest
+    @ArgumentsSource(TestCaseProvider::class)
+    fun `test Javadoc output`(testCase: TestCase) {
+        assumeTrue(testCase.outputsJavadoc)
+
+        testDokkaOutput(
+            testCase = testCase,
+            format = "javadoc",
+        )
+    }
+
+    private fun testDokkaOutput(
+        testCase: TestCase,
+        format: String,
+    ) {
+        val expectedDataDir = testCase.expectedDataDir.resolve(format)
+        val dokkaOutputDir = testCase.dokkaOutputDir.resolve(format)
+
+        assert(expectedDataDir.exists() && expectedDataDir.isDirectory()) {
+            "Missing expectedDataDir: ${expectedDataDir.toUri()}"
+        }
+
+        testCase.project.runner
+            .addArguments(
+                testCase.dokkaGenerateTask,
+                "--stacktrace",
+            )
+            .build {
+                dokkaOutputDir.shouldBeADirectory()
+
+                withClue({
+                    """
+                    expectedDataDir: ${expectedDataDir.toUri()}
+                    actualOutputDir: ${dokkaOutputDir.toUri()}
+                    """.trimIndent()
+                }) {
+                    withClue("expect file trees are the same") {
+                        val expectedFileTree = expectedDataDir.toTreeString()
+                        val actualFileTree = dokkaOutputDir.toTreeString()
+                        expectedFileTree shouldBe actualFileTree
```

Comment:
This test will be somewhat flaky if you don't filter dot files (or filter files per `.gitignore`). E.g. Poking around test fixtures with Finder on macOS will create a bunch of `.DS_Store` files which will fail this test when run locally

Reply:
Trying to filter out some files seems like a lot of work, and for something that might only happen sometimes, and can be resolved by manually deleting the files.

Reply:
I disagree; false positive test failures add a lot of DevX friction. I feel it shouldn't be hard to filter out at least the `.DS_Store` files

Reply:
> I disagree; false positive test failures add a lot of DevX friction.

Yes, false positive test failures add friction, but there haven't been any false positive failures.